index.ts

```diff
@@ -54,112 +62,111 @@ const BUFFER_LENGTH = ONE_SECOND * BUFFER_LENGTH_SECONDS;
+  const transcriptions = conversation.transcriptions.map(({ text, startByte, endByte, precannedLLMResponse }) => ({ text, startByte, endByte, precannedLLMResponse }));
+}
+
 // big bag of state
 // TODO: Roll data off
 // Implement canonical dialogue
 const conversation: Conversation = {
   audioBuffer: Buffer.alloc(0),
   transcriptions: [],
-  precannedLLMResponses: [],
-  canonicalDialogue: []
+  canonicalDialogue: [],
+  watermark: 0,
+}
+
+const seedCanonicalDialogue = () => {
+  conversation.canonicalDialogue.push({ speaker: 'agent', response: "Hey yacine! long time no see. How have you been?" });
+  conversation.canonicalDialogue.push({ speaker: 'user', response: "Alice! It's been ages. I've been good, quite busy with work though. And you?" });
+  conversation.canonicalDialogue.push({ speaker: 'agent', response: "Pretty much the same here. Busy, busy. I've started reading manga and watahcing anime, it's really therapeutic." });
 }
+seedCanonicalDialogue();
 
 audioProcess.stdout.on('readable', () => {
   let data;
-  // Implicit, should be injected into conversation
   while (data = audioProcess.stdout.read()) {
     addAudioDataToConversation(conversation, data);
   }
 });
 
-// TODO: Renable resposne reflex, and abstract away
-// readline.emitKeypressEvents(process.stdin);
-// process.stdin.setRawMode(true);
-// process.stdin.on('keypress', async (str, key) => {
-//   // Detect Ctrl+C and manually emit SIGINT to preserve default behavior
-//   if (key.sequence === '\u0003') {
-//     // The reason I wait for these promises is because ggml core dumps when you do it in the middle of an infer
-//     // Yes, lurk
-//     if (globalLlamaPromise) {
-//       await globalLlamaPromise;
-//     }
-//     if (globalWhisperPromise) {
-//       await globalWhisperPromise;
-//     }
-//     process.exit();
-//   }
-
-//   // R for respond
-//   if (key.sequence === 'r') {
-//     for (let i = conversation.transcriptions.length - 1; i >= 0; i--) {
-//       const transcription = conversation.transcriptions[i];
-//       const {startByte, text, responses } = transcription;
-//       if (responses.length > 0) {
-//         // TODO: simulate response
-//       }
-//     }
-//   }
-// });
 
 const whisperLoop = async () => {
   // TODO: Figure out why segfault occurs when no waiting
   // Likely a hint in the original whisper cpp napi addon code
   // I presume the cause is init func early return?
   await new Promise(r => setTimeout(r, 1000));
-
   let start = 0;
   while (true) {
     const total_buffer_length = conversation.audioBuffer.length;
-    const audioSlice = conversation.audioBuffer.slice(start, total_buffer_length);
+    const audioSlice = conversation.audioBuffer.slice(Math.max(start, conversation.watermark), total_buffer_length);
+    globalWhisperPromise = whisper.whisperInferenceOnBytes(audioSlice);
+    // const [result] = await Promise.all([globalWhisperPromise, globalLlamaPromise]);
```

Comment:
[nit] I dislike commented out code, either fix the feature or remove the code, but otherwise it introduces code which others must tip-toe around, not knowing if it should be there, or if it is even relevant anymore

Reply:
yeah i oopsie